common/src/main/java/com/revenuecat/purchases/common/Backend.kt

```diff
@@ -346,17 +350,64 @@ class Backend(
             }
         }
         synchronized(this@Backend) {
-            identifyCallbacks.addCallback(call, cacheKey, onSuccessHandler to onErrorHandler)
+            identifyCallbacks.addCallback(call, dispatcher, cacheKey, onSuccessHandler to onErrorHandler)
         }
     }
 
-    @Suppress("UnusedPrivateMember") // WIP: Remove suppress
     fun postDiagnostics(
         diagnosticsList: List<JSONObject>,
         onSuccessHandler: (JSONObject) -> Unit,
         onErrorHandler: (PurchasesError, Boolean) -> Unit
     ) {
-        // WIP: Perform request
+        val cacheKey = diagnosticsList.map { it.hashCode().toString() }
+
+        val body = mapOf("entries" to JSONArray(diagnosticsList))
+        val call = object : Dispatcher.AsyncCall() {
+            override fun call(): HTTPResult {
+                return httpClient.performRequest(
+                    appConfig.dianosticsURL,
+                    "/telemetry",
+                    body,
+                    authenticationHeaders
+                )
+            }
+
+            override fun onError(error: PurchasesError) {
+                synchronized(this@Backend) {
+                    diagnosticsCallbacks.remove(cacheKey)
+                }?.forEach { (_, onErrorHandler) ->
+                    onErrorHandler(error, error.code == PurchasesErrorCode.NetworkError)
+                }
+            }
+
+            override fun onCompletion(result: HTTPResult) {
+                synchronized(this@Backend) {
+                    diagnosticsCallbacks.remove(cacheKey)
+                }?.forEach { (onSuccessHandler, onErrorHandler) ->
+                    if (result.isSuccessful()) {
+                        onSuccessHandler(result.body)
+                    } else {
+                        val error = result.toPurchasesError()
+                        val shouldRetry = result.responseCode >= RCHTTPStatusCodes.ERROR ||
+                            error.code == PurchasesErrorCode.NetworkError
+                        onErrorHandler(error, shouldRetry)
+                    }
+                }
+            }
+        }
+        synchronized(this@Backend) {
+            diagnosticsCallbacks.addCallback(call, diagnosticsDispatcher, cacheKey, onSuccessHandler to onErrorHandler)
+        }
+    }
+
+    private fun enqueue(
+        call: Dispatcher.AsyncCall,
+        dispatcher: Dispatcher,
+        randomDelay: Boolean = false
+    ) {
+        if (!dispatcher.isClosed()) {
```

Comment:
Is it safe to ignore this if it is closed?

Reply:
Hmm I just moved this method here without changing it. The dispatcher should only be closed when calling `Purchases.close()`, which would probably only be used if a user wants to reconfigure the SDK (not a common scenario). But you're right, we probably want to log something if not.